### PR TITLE
V3 conformant

### PR DESCRIPTION
**Pull request by awharn**
**What It Does**

Updates the DB2 plug-in for V3 conformance

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: src/cli/call/procedure/Procedure.handler.ts [hunk 1]
```diff
@@ -55,6 +55,7 @@ export default class ProcedureHandler extends DB2BaseHandler {
 
         // Return as an object when using --response-format-json
         params.response.data.setObj(response);
+        params.response.data.setExitCode(response.success ? 0 : 1);
```
Comment (zFernand0) [inline src/cli/call/procedure/Procedure.handler.ts:58]: If every command handler extends `DB2BaseHandler`, could we move this line over there? 😋 

And maybe have a try/catch on the `DB2BaseHandler` to catch any imperative errors, set the exit code and then proceed to throw the error?

Technically speaking though, when an error is thrown, the exit code is should be automatically set to 1 (by the Imperative) 😋

File: package.json [hunk 2]
```diff
@@ -28,49 +28,44 @@
...
   "imperative": {
     "configurationModule": "lib/imperative.js"
   },
   "dependencies": {
-    "ibm_db": "3.2.2"
+    "ibm_db": "3.2.3"
   },
   "peerDependencies": {
-    "@zowe/imperative": "^5.0.0"
+    "@zowe/imperative": ">8.0.0-next.0 <9.0.0"
```
Comment (zFernand0) [inline package.json:43]: I think we have gone back and forth on this, but curious if the intention is to allow for any `8.x.x` (non-prerelease included) to be compatible with this version. 🤔

**Review by zFernand0 — COMMENTED** [review]
PR LGTM! 😋 
left a few questions/comments 😋

File: package.json (hunk 2 above)
Comment (awharn) [reply package.json:43]: Not really, I plan for this to update when we are out of prerelease to be >8.0.0.

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
Thanks for addressing the feedback! 🥳

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,11 @@
 
 All notable changes to the IBM® Db2® Plug-in for Zowe CLI will be documented in this file.
 
+## Recent Changes
+
+- BugFix: Exit code 1 when DB2 calls do not succeed
```
Comment (anaxceron) [inline CHANGELOG.md:7]: We should explain the actual fix for this. Also, this needs a PR/issue link.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,11 @@
...
+- V3 Breaking: Updated engines, removed health check and v1 profile support, added port and database to template
```
Comment (anaxceron) [inline CHANGELOG.md:8]: This needs a PR/issue link.

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,11 @@
...
+
 ## `5.0.6`
 
 - BugFix: Update follow-redirects transitive dependency to resolve technical debt
```
Comment (anaxceron) [inline CHANGELOG.md:12]: This needs a PR/issue link.

File: README.md [hunk 6]
```diff
@@ -102,18 +102,30 @@ When an unsuccessful message displays, you can troubleshoot the installation by
...
-    zowe profiles create db2 <profile name> -H <host> -P <port> -u <user> -p <password> -d <database>
+    zowe config init
+    ```
+3.  Set the port number to your Db2 port:
+
+    ```
+    zowe config set profiles.db2.properties.port <port number>
+    ```
+
+4.  Set the database to your Db2 database:
```
Comment (anaxceron) [inline README.md:117]: data base should be two words.

File: README.md [hunk 7]
```diff
@@ -102,18 +102,30 @@ When an unsuccessful message displays, you can troubleshoot the installation by
...
+
+    ```
+    zowe config set profiles.db2.properties.database <database name>
+    ```
+
+5. If required, set the SSL File to the CA certificate used for the DB2 server
```
Comment (anaxceron) [inline README.md:123]: this should end w/ a colon

**Review by anaxceron — APPROVED** [review]
Approving this but made some comments for minor edits.

File: CHANGELOG.md (hunk 4 above)
Comment (awharn) [reply CHANGELOG.md]: Added

File: CHANGELOG.md (hunk 5 above)
Comment (awharn) [reply CHANGELOG.md]: Added

File: README.md (hunk 6 above)
Comment (awharn) [reply README.md:117]: I do not believe database should be two words. It is referenced as `database` in Zowe Docs.

File: README.md (hunk 7 above)
Comment (awharn) [reply README.md]: Fixed

File: CHANGELOG.md (hunk 3 above)
Comment (awharn) [reply CHANGELOG.md]: Updated link and wording to state that the command will exit with code 1 if something goes wrong with issuing commands. This tells automation that something went wrong.

**Review by traeok — APPROVED** [review]
LGTM - thanks @awharn! 😁

**Pull request merged**